primitives/session/src/lib.rs

```diff
@@ -106,6 +106,23 @@ impl GetValidatorCount for MembershipProof {
 	}
 }
 
+/// Trait for retrieving the session info needed for online node inspection.
+///
+/// This trait is used for decouple the pallet-session dependency from im-online
+/// module so that the user of im-online & offences modules can pass any list of
+/// validators that are considered to be online in each session, particularly useful
+/// for the Substrate-based projects having their own staking implementation
+/// instead of using pallet-staking directly.
+pub trait SessionInterface<ValidatorId> {
```

Comment:
This name should be re-worked. We already have a `SessionInterface` trait defined in staking and given the near proximity it'll be confusing.